app/src/main/java/io/neurolab/NFBServer.java

```diff
@@ -0,0 +1,79 @@
+package io.neurolab;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.locks.ReentrantLock;
+
+import io.neurolab.model.Task;
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+public class NFBServer extends NeuroSettings implements DataReceiver {
+
+    private long currentTimestamp;
+    public static int samplesPerSecond = 256;
+    public static int numChannels = 2;
+
+    private List<double[]> inputData;
+    protected ConcurrentLinkedDeque<double[]> currentData = new ConcurrentLinkedDeque<double[]>();
+    private int numSamples;
+    private static ReentrantLock lock = new ReentrantLock();
+    public int minimumNewSamples = 16;
+    private LinkedList<Task> tasks;
+    private long lastTimestamp;
```

Comment:
Where is being used?

Reply:
Check out the append data method. Currently we are only initializing this there, but we will use it in the successive PRs directed to update this class once FileOutputTask get's successfully created without any issue. Removing this right now won't have much effect as we need to add it in the very near future again then. I have kept this to finalize the appendData method to the max extent.

Reply:
@jddeep ok cool